astropy/time/core.py

```diff
@@ -2276,6 +2284,72 @@ def __add__(self, other):
     def __radd__(self, other):
         return self.__add__(other)
 
+    def mean(
+            self: 'Time',
```

Comment:
See other comment on 'Time'. But also, `self` generally doesn't need to be type hinted.